lms/djangoapps/commerce/tests/__init__.py

```diff
@@ -99,3 +100,44 @@ def test_client_unicode(self):
         )
         actual_object = ecommerce_api_client(self.user).baskets(1).order.get()
         self.assertEqual(actual_object, {u"result": u"Préparatoire"})
+
+    @httpretty.activate
+    @freeze_time('2015-7-2')
+    @override_settings(JWT_ISSUER='http://example.com/oauth', JWT_EXPIRATION=30)
+    def test_tracking_context_with_worker_user(self):
```

Comment:
From the looks of it, you're adding all this to test your change to [API client initialization](https://github.com/edx/edx-platform/pull/11759/files#diff-29385ceb5d39ae786ef9c70a4d7c6b58R35). Instead of copying `test_tracking_context` wholesale, be more precise about what you're testing. All you need to verify is that you can initialize an API client for a user without a profile.